### PR TITLE
Feature/board section implement

### LINKED ISSUE #31: iOS Feature - UIUX x Data in Board Section of Explore module
# Feature

### Description
Implement the feature in the `Board` section of the `Explore` module.

### Tasks
- [x]  Static UI implement
- [x]  Static UI optimization
- [x]  User's interaction handle

### PR DESCRIPTION
**Pull request by Vince-Ku**
fix #31 

Please review this PR after #39 , I'll rebase this branch onto `Main` branch once #39 is merged.

### CODE REVIEW

File: SCSH_3C_iOS/SCSH_3C_iOS/Repository/Model/Theme.swift [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+//
+
+import Foundation
+
+protocol ThemeType {}
+
+enum HomeTheme: ThemeType {
+    case Explore
+    case Subject
+    case _3C
```
Comment (shadow3x3x3) [inline SCSH_3C_iOS/SCSH_3C_iOS/Repository/Model/Theme.swift]: Does this qualify as a violation of naming conventions? 🤣 

How about using `Product3C` instead?

File: SCSH_3C_iOS/SCSH_3C_iOS/Modules/Home/View/HomeViewController.swift [hunk 2]
```diff
@@ -107,6 +85,46 @@ class HomeViewController: UIViewController {
...
     }
+    
+    private func bindEvents() {
+        viewModel.themes
+            .zip(viewModel.firstHeaderIndex)
+            .throttle(for: .milliseconds(100), scheduler: DispatchQueue.main, latest: true)
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] _, firstIndex in
+                self?.headerCollectionView.reloadData()
+                self?.bodyCollectionView.reloadData()
```
Comment (shadow3x3x3) [inline SCSH_3C_iOS/SCSH_3C_iOS/Modules/Home/View/HomeViewController.swift]: I'm not sure, but it seems like a sequence of `reloadData()`.

What side effects or conflicts might arise if we use *Async* for handling their respective data?

**Review by shadow3x3x3 — COMMENTED** [review]
Please have a look.

File: SCSH_3C_iOS/SCSH_3C_iOS/Repository/Model/Theme.swift (hunk 1 above)
Comment (Vince-Ku) [reply SCSH_3C_iOS/SCSH_3C_iOS/Repository/Model/Theme.swift]: Agree !! I don't know why I do that 🤣

> How about using Product3C instead?

ok !

File: SCSH_3C_iOS/SCSH_3C_iOS/Modules/Home/View/HomeViewController.swift (hunk 2 above)
Comment (Vince-Ku) [reply SCSH_3C_iOS/SCSH_3C_iOS/Modules/Home/View/HomeViewController.swift]: **Based on the conditions below I wrote the reloadData() in the same block.**

1. `headerCollectionView` and `bodyCollectionView` share the same data source.
2.  neither can be updated without the other, or it is ineffective.

>  if we use Async for handling their respective data?

I can't understand this one, could you describe more detail?

File: SCSH_3C_iOS/SCSH_3C_iOS/Repository/Model/Theme.swift (hunk 1 above)
Comment (Vince-Ku) [reply SCSH_3C_iOS/SCSH_3C_iOS/Repository/Model/Theme.swift]: [style: renaming HomeTheme](https://github.com/No-57/SCSH_3C_iOS/pull/32/commits/233b01545e2d98edee698ae18843490b31970a95)

File: SCSH_3C_iOS/SCSH_3C_iOS/Modules/Home/View/HomeViewController.swift (hunk 2 above)
Comment (shadow3x3x3) [reply SCSH_3C_iOS/SCSH_3C_iOS/Modules/Home/View/HomeViewController.swift]: If there's a dependency, that's fine.

I assumed these could be two separate standalone tasks, at first. In that case, can run them asynchronously.

**Review by shadow3x3x3 — APPROVED** [review]
LGTM

**Pull request merged**